src/test/cypress/integration/exam/ExamDateVerification.spec.ts

```diff
@@ -120,17 +119,16 @@ describe('Exam date verification', () => {
                         courseManagementRequests.generateMissingIndividualExams(exam);
                         courseManagementRequests.prepareExerciseStartForExam(exam);
                         cy.login(student, `/courses/${course.id}/exams`);
-                        cy.contains(exam.title).click();
-                        cy.contains('Welcome to ' + exam.title).should('be.visible');
+                        courseOverview.openExam(exam.id);
+                        cy.contains(exam.title).should('be.visible');
                         examStartEnd.startExam();
-                        cy.contains(textExercise.title).should('be.visible').click();
+                        examNavigationBar.openExerciseAtIndex(0);
                         cy.fixture('loremIpsum.txt').then((submissionText) => {
                             textEditor.typeSubmission(submissionText);
                         });
-                        cy.contains('Save').click();
+                        examNavigationBar.clickSave();
                         cy.contains('This is the end of ' + exam.title, { timeout: 40000 });
```

Comment:
Here it might make sense to give the element an id and then only assert that the exam title is shown in it.
Something like this
```suggestion
                        cy.get('#exam-finished-message', { timeout: 40000 }).should('contain.text', exam.title);
```